charts/consul/templates/connect-inject-deployment.yaml

```diff
@@ -107,7 +107,8 @@ spec:
                 {{- else }}
                 -transparent-proxy-default-overwrite-probes=false \
                 {{- end }}
-                {{- if (and .Values.dns.enabled .Values.dns.enableRedirection)}}
+                {{- if (and .Values.dns.enabled .Values.dns.enableRedirection) }}
+                -release-prefix={{ template "consul.fullname" . }} \
```

Comment:
I think we should always set `-resource-prefix` because other developers might assume it's available.

Also I see there's a `CONSUL_FULLNAME` above. This is remnants from when we used to have a resource prefix flag I'm pretty sure. It can be removed.